src/KineticConvolution/Direction.cs

```diff
@@ -31,12 +31,132 @@ public Direction(IAlgebraicNumber x, IAlgebraicNumber y)
 
         public IAlgebraicNumber Y { get; }
 
-        public bool Equals(Direction d)
+        /// <inheritdoc />
+        public bool Equals(Direction? other)
         {
-            return
-                DirectionHelper.Determinant(this, d).IsZero() &&
-                X.Sign() == d.X.Sign() &&
-                Y.Sign() == d.Y.Sign();
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            return DirectionHelper.Determinant(this, other).IsZero()
+                && X.Sign() == other.X.Sign()
+                && Y.Sign() == other.Y.Sign();
+        }
+
+        public int Compare(Direction direction1, Direction direction2)
+        {
+            if (direction1.Equals(direction2))
+            {
+                return 0;
+            }
+
+            return direction1.BelongsTo(
+                       new DirectionRange(
+                           this,
+                           direction2,
+                           Orientation.CounterClockwise))
+                       ? 1
+                       : -1;
+        }
+
+        public Direction FirstOfCounterClockwise(
+            Direction direction1,
+            Direction direction2) =>
+            Compare(direction1, direction2) switch
+            {
+                -1 => direction2,
+                1 => direction1,
+                0 => direction1,
+                _ => throw new NotSupportedException() // TODO add a message
+            };
+
+        public Direction LastOfCounterClockwise(
+            Direction direction1,
+            Direction direction2) =>
+            Compare(direction1, direction2) switch
+            {
+                -1 => direction1,
+                1 => direction2,
+                0 => direction1,
+                _ => throw new NotSupportedException() // TODO add a message
+            };
+
+        public bool BelongsToShortestRange(DirectionRange directions)
+        {
+            var s = DirectionHelper.Determinant(directions.Start, directions.End);
+            if (s.IsStrictlyPositive())
+            {
+                return
```

Comment:
this return IsStrictlyTheSame() return as below? ^^

Reply:
corrigé 👍